apis/v1alpha2/shared_types.go

```diff
@@ -401,3 +430,86 @@ type AnnotationKey string
 // +kubebuilder:validation:MinLength=0
 // +kubebuilder:validation:MaxLength=4096
 type AnnotationValue string
+
+// AddressRouteMatches defines AddressMatch rules for inbound traffic according to
+// source and/or destination address of a packet or connection.
+type AddressRouteMatches struct {
+	// SourceAddresses indicates the originating (source) network
+	// addresses which are valid for routing traffic.
+	//
+	// Support: Extended
+	SourceAddresses []AddressMatch `json:"sourceAddresses"`
```

Comment:
I'm suspicious of the need for this.  It feels like firewall behavior, but it doesn't purport to drop non-matching traffic, just to ignore it, IIUC?  That seems potentially problematic for some impls (e.g. iptables would not match and allow the packet to continue for further processing).

Can you help me understand what it is for?

Reply:
Good question, it's been a while since we reviewed this, and I can't remember the specific use cases for source IP matching, maybe @shaneutt does?

Reply:
Discussed in community meeting and it seems like we may need to rethink this a bit, more discussion in #1127.

Reply:
This has been removed from the API while we reconsider how it could work.